src/SetupApi/SetupApi.cs

```diff
@@ -219,6 +219,272 @@ public static partial class SetupApi
+        /// A pointer to an <see cref="SP_DEVINFO_DATA"/> structure that specifies the device information element in <paramref name="deviceInfoSet"/>.
+        /// This parameter is optional and can be <see langword="null"/>. If this parameter is specified, <see cref="SetupDiSetSelectedDriver"/> sets,
+        /// or resets, the selected driver for the specified device. If this parameter is <see langword="null"/>, <see cref="SetupDiSetSelectedDriver"/>
+        /// sets, or resets, the selected class driver for <paramref name="deviceInfoSet"/>.
+        /// </param>
+        /// <param name="driverInfoData">
+        /// <para>
+        /// An <see cref="SP_DRVINFO_DATA"/> structure that specifies the driver to be selected. This parameter is optional and can be <see langword="null"/>.
+        /// If this parameter and <paramref name="deviceInfoData"/> are supplied, the specified driver must be a member of a driver list that is associated
+        /// with <paramref name="deviceInfoData"/>. If this parameter is specified and <paramref name="deviceInfoData"/> is <see langword="null"/>, the
+        /// driver must be a member of the global class driver list for <paramref name="deviceInfoSet"/>. If this parameter is <see langword="null"/>, the
+        /// selected driver is reset for the device information element, if <paramref name="deviceInfoData"/> is specified, or the device information set,
+        /// if <paramref name="deviceInfoData"/> is <see langword="null"/>.
+        /// </para>
+        /// <para>
+        /// If the <see cref="SP_DRVINFO_DATA.Reserved"/> is <see langword="null"/>, the caller is requesting a search for a driver node with the specified
+        /// parameters (<see cref="SP_DRVINFO_DATA.DriverType"/>, <see cref="SP_DRVINFO_DATA.Description"/>, and <see cref="SP_DRVINFO_DATA.ProviderName"/>).
+        /// If a match is found, that driver node is selected. The <see cref="SP_DRVINFO_DATA.Reserved"/> field is updated on output to reflect
+        /// the actual driver node where the match was found. If a match is not found, the function fails and a call to <see cref="Marshal.GetLastWin32Error"/>
+        /// returns <c>ERROR_INVALID_PARAMETER</c>.
+        /// </para>
+        /// </param>
+        /// <returns>
+        /// The function returns <see langword="true"/> if it is successful.
+        /// Otherwise, it returns <see langword="false"/> and the logged error can be retrieved with a call to <see cref="Marshal.GetLastWin32Error"/>.
+        /// </returns>
+        /// <seealso href="https://msdn.microsoft.com/en-us/library/windows/hardware/ff552183(v=vs.85).aspx"/>
+        [DllImport(nameof(SetupApi), SetLastError = true, CharSet = CharSet.Unicode)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        public static extern bool SetupDiSetSelectedDriver(
+            SafeDeviceInfoSetHandle deviceInfoSet,
+            ref SP_DEVINFO_DATA deviceInfoData,
+            ref SP_DRVINFO_DATA driverInfoData);
```

Comment:
Change to:

```cs
[Friendly(FriendlyFlags.In | FriendlyFlags.Optional)] SP_DEVINFO_DATA* deviceInfoData
[Friendly(FriendlyFlags.Bidirectional | FriendlyFlags.Optional)] SP_DRVINFO_DATA* driverInfoData
```

This allows each parameter to be optional (as documented) while still generated friendly overloads (even friendlier than you've got here).